vllm/v1/core/scheduler.py

```diff
@@ -61,9 +62,11 @@ def __init__(
         # Priority queues for requests.
         self.waiting: Deque[Request] = deque()
         self.running: List[Request] = []
-        # The requests that have been scheduled and are being executed
-        # by the executor.
-        self.scheduled_req_ids: Set[str] = set()
+        # req_id -> Number of times the request has been scheduled.
+        # We can only schedule a request more then once before the previous
+        # scheduling step is finished when PP is enabled and the request
+        # prompt is chunked.
```

Comment:
I find this comment a bit difficult to parse. What about this?

With PP, when the input prompt is divided into chunks, we can schedule a new chunk even before the previous chunk has completed the full pipeline stages. This helps reduce TTFT.

Reply:
SGTM. Will update.